Dev/config/FirebaseFunctions.js

```diff
@@ -320,16 +320,18 @@ export default class FirebaseFunctions {
 
         let currentClass = await this.getClassByID(classID);
         let arrayOfStudents = currentClass.students;
-        let updatedArrayOfStudents = arrayOfStudents.map((student) => {
-
+        arrayOfStudents.forEach((student, index) => {
             //If the attendance already exists, then the code will automatically replace
             //the old attendance with this one
-            student.attendanceHistory[selectedDate] = (absentStudents.includes(student.ID) ? false : true);
-
+            let copyOfStudent = student;
+            let { attendanceHistory } = copyOfStudent;
+            attendanceHistory[selectedDate] = (absentStudents.includes(student.ID) ? false : true);
+            copyOfStudent.attendanceHistory = attendanceHistory;
+            arrayOfStudents[index] = copyOfStudent;
```

Comment:
This is fine for now. Ideally we should try to avoid doing this expensive processing in the client and instead have the schema returns as close to what we want as possible. But this is fine for now,